2 main things: score counter and game simulator with names and scores

Fabbernat opened the pull request:
Refactor Settings and GameMode handling for better consistency and Enum support

- Converted GameMode to an Enum for consistency with MapSizes and MapTypes, allowing the use of .name attributes.
- Updated Settings to ensure GAME_MODE and MAP_SIZE are compatible with .name-based access.
- Enhanced the
un_game.py script to handle map_settings dynamically using .name for Enum members and raw values where appropriate.
- Fixed the ModuleNotFoundError by addressing Python package structure and import paths in the polytopia_game module.

This refactor improves readability, robustness, and consistency across game settings.

Fix: Properly initialize and populate players in the game

- Ensured that the `self.players` list is properly initialized with player names.
- Corrected the issue where `player_scores` was not populated due to an empty `self.players` list.
- Added a check to verify that `self.players` contains valid player names before assigning scores.
- Ensured that player names are shuffled and linked to the players correctly for dynamic gameplay.
- Improved the readability and robustness of the `run()` method by ensuring proper initialization and shuffling logic.

This resolves the issue where `player_scores` was not populated, leading to empty results in the game run.

Fabbernat committed:
Add score reward system with actions and calculations

Fabbernat committed:
Install Pillow library for image processing
Add Pillow dependency and update map_renderer.py imports

Fabbernat committed:
Subject: [PATCH] Install Pillow library for image processing
Add Pillow dependency and update map_renderer.py imports
---
Index: polytopia_score_counter/rewards.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/polytopia_score_counter/rewards.py b/polytopia_score_counter/rewards.py
--- a/polytopia_score_counter/rewards.py	(revision 76cc9a78e3f9c0ec05c8efe9be1182d1d84151c4)
+++ b/polytopia_score_counter/rewards.py	(date 1736350762769)
@@ -52,4 +52,7 @@
     if action == "discover_fog":
         return 5

+    if action == "park":
+        return 250
+
     return 0  # Default reward for neutral actions
Index: polytopia_score_counter/actions.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/polytopia_score_counter/actions.py b/polytopia_score_counter/actions.py
--- a/polytopia_score_counter/actions.py	(revision 76cc9a78e3f9c0ec05c8efe9be1182d1d84151c4)
+++ b/polytopia_score_counter/actions.py	(date 1736351187218)
@@ -11,5 +11,22 @@
     "gain_territory",
     "lose_territory",
     "research_tech",
-    "discover_fog"
+    "discover_fog",
+    "park",
+]
+
+_2D_ACTIONS = [
+    {"action": "train_unit", "unit_type": "archer"},
+    {"action": "lose_unit", "unit_type": "warrior"},
+    {"action": "upgrade_city", "city_population_gain": 3},
+    {"action": "place_structure", "structure": "monument"},
+    {"action": "capture_city", "city_population_gain": 1},
+    {"action": "lose_city", "city_population_loss": 2},
+    {"action": "gain_territory"},
+    {"action": "lose_territory"},
+    {"action": "research_tech", "tech_tier": 1},
+    {"action": "research_tech", "tech_tier": 2},
+    {"action": "research_tech", "tech_tier": 3},
+    {"action": "discover_fog"},
+    {"action": "park"},
 ]
Index: main.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/main.py b/main.py
--- a/main.py	(revision 76cc9a78e3f9c0ec05c8efe9be1182d1d84151c4)
+++ b/main.py	(date 1736351187218)
@@ -1,18 +1,20 @@
-# main.py
-
 from polytopia_score_counter.rewards import calculate_reward
-from polytopia_score_counter.actions import ACTIONS
+from polytopia_score_counter.actions import _2D_ACTIONS
+

 def main():
-    # Test reward calculations
-    print("Reward for training an archer:", calculate_reward("train_unit", unit_type="archer"))
-    print("Reward for losing a warrior:", calculate_reward("lose_unit", unit_type="warrior"))
-    print("Reward for upgrading a city with 3 population gain:", calculate_reward("upgrade_city", city_population_gain=3))
-    print("Reward for placing a monument:", calculate_reward("place_structure", structure="monument"))
-    print("Reward for researching a tier 2 technology:", calculate_reward("research_tech", tech_tier=2))
-    print("Reward for discovering a fog tile:", calculate_reward("discover_fog"))
-    print("Reward for gaining a territory tile:", calculate_reward("gain_territory"))
-    print("Reward for losing a city with 2 population:", calculate_reward("lose_city", city_population_loss=2))
+    score = 0
+
+    # Loop through each action and calculate the reward
+    for action_info in _2D_ACTIONS:
+        action = action_info["action"]
+        # Call calculate_reward dynamically with the appropriate arguments
+        reward = calculate_reward(action, **{key: value for key, value in action_info.items() if key != "action"})
+        score += reward
+        print(f"Reward for {action}: {reward}")
+
+    print(f'Total score: {score}')
+

 if __name__ == "__main__":
     main()

Fabbernat committed:
Refactor Settings and GameMode handling for better consistency and Enum support

- Converted GameMode to an Enum for consistency with MapSizes and MapTypes, allowing the use of .name attributes.
- Updated Settings to ensure GAME_MODE and MAP_SIZE are compatible with .name-based access.
- Enhanced the 
un_game.py script to handle map_settings dynamically using .name for Enum members and raw values where appropriate.
- Fixed the ModuleNotFoundError by addressing Python package structure and import paths in the polytopia_game module.

This refactor improves readability, robustness, and consistency across game settings.

Fabbernat committed:
Fix: Properly initialize and populate players in the game

- Ensured that the `self.players` list is properly initialized with player names.
- Corrected the issue where `player_scores` was not populated due to an empty `self.players` list.
- Added a check to verify that `self.players` contains valid player names before assigning scores.
- Ensured that player names are shuffled and linked to the players correctly for dynamic gameplay.
- Improved the readability and robustness of the `run()` method by ensuring proper initialization and shuffling logic.

This resolves the issue where `player_scores` was not populated, leading to empty results in the game run.

DylanHannaScience commented:
Hello! 
Thanks so much for this amazing MR, I can see you've put a lot of thought into it 😁 
As you can probably tell, I'm not really maintaining this code (I actually thought this was a private repo 😅), so I'm more than happy to hand the repo over to you if you want to take it and become the owner.

It's really cool to see that you saw some value in this project and wanted to contribute 😄

Fabbernat committed:
Solves: I want to use MVC in my flask app. What are the conventions? Where to put the Views, models, controllers? WHeret to put static files like css and img? Is there a wwwroot or static or templates folder?

Fabbernat committed:
Add basic Flask app structure with navigation routes

- Set up a Flask app to serve web pages.
- Added routes for:
  - Home (`/`)
  - Privacy (`/privacy`)
  - Dashboard (`/dashboard`)
  - About (`/about`)
  - Settings (`/settings`)
  - Hall of Fame (`/hall_of_fame`)
  - Throne Room (`/throne_room`)
- Created navigation between pages to enable smooth browsing.
- Implemented basic HTML responses to test routing functionality.

This commit lays the foundation for serving dynamic content and creating a user-friendly web interface.

Fabbernat committed:
Implement basic Minimax functions for Polytopia AI

- Add `vegallapot(n)`: Checks if the game has reached a terminal state.
- Add `hasznossag(n)`: Evaluates game states based on city and unit count.
- Add `n_szomszedai(n)`: Generates possible next states for the current player.
- Basic mechanics include unit training and attacking.
- Lays groundwork for reinforcement learning integration.

Fabbernat committed:
initial Java commit to this PYthon projevt

Fabbernat committed:
v1.0 cimple console app works, with no particular game logic implemented

Fabbernat committed:
v1.1: significantly improve project structure, still runs all well and fine

Fabbernat committed:
v1.2: implemented new game maker logic with ID's. Can make any amount of id'd games

Fabbernat committed:
v1.3: HUGE update: add guids, pre-stored games TODO: store and delete newly created games

Fabbernat committed:
v1.3: move resources into a folder dedicated for them

Fabbernat committed:
v1.4: Fixed a lot of bugs in reading in guidReference.txt and games.txt, no errors or exeptions thrown in this version

Fabbernat committed:
v1.5: games now can also be queried & listed with the "gam" or "games" command.

Fabbernat committed:
v1.5.1: add missing whitespace

Fabbernat committed:
Revise README for clarity and new contributions

Updated README to reflect changes in contributions and added information about the Java console app.

fabbernatvasvari committed:
Fix games bug

fabbernatvasvari committed:
Fixed some  bugs and improved code quality

fabbernatvasvari committed:
Now all game names start with a Capital Letter.

fabbernatvasvari committed:
TODO: fix problems of this patch

fabbernatvasvari committed:
updated&improved `help` command

fabbernatvasvari committed:
Add all 12 base tribes' basic stats

fabbernatvasvari committed:
beuatiful 16x16 tiles blue&green grid works, integrated with the cionsole app as well, so pretty good so far

fabbernatvasvari committed:
v1.6: add Perlin noise via PerlinNoise.java

fabbernatvasvari committed:
v1.6.1: I am pretty satisfied with the algorithm so far

fabbernatvasvari committed:
v1.6.2: improved logic, capitals now don't spawn at the edge of the map

fabbernatvasvari committed:
v1.6.3: balance changes

fabbernatvasvari committed:
v1.6.3.1: better colors

fabbernatvasvari committed:
v1.6.3.2: add village-filling scratch

fabbernatvasvari committed:
v1.6.3.2: Village generation now also works, although neighbor cities still needs to be fixed

fabbernatvasvari committed:
v1.6.3.3: 35 is pretty much the lower limit of how many villages can spawn. It does not always work, but it gives decent results when succeeds

fabbernatvasvari committed:
v1.6.3.3.1: same

fabbernatvasvari committed:
v1.6.3.3.1: increased village spawn logic, and I dont think it will fail to generate with this algorithm

fabbernatvasvari committed:
v1.6.3.3.2: good version for lakes

fabbernatvasvari committed:
v1.6.3.3.3: simulate bardur gameplay

fabbernatvasvari committed:
v1.6.3.3.4: simulate bardur gameplay even on large maps

fabbernatvasvari committed:
v1.6.3.3.5: every map is drylands right now

fabbernatvasvari committed:
v1.6.3.3.6: save before messup

fabbernatvasvari committed:
v1.6.3.3.7: nullPointerException

fabbernatvasvari committed:
v1.6.3.3.8: IT WORKS!!! Finally, it works! TODO: WW runs into infinite loop, fix that. Then, add shallow water, ruins and valid village spawns

fabbernatvasvari committed:
v1.6.3.3.9: first version after IT WORKS!!! Finally, it works! TODO: WW runs into infinite loop, fix that. Then, add shallow water, ruins and valid village spawns

fabbernatvasvari committed:
v1.6.3.3.10: mountains do not spawn any more on ocean

fabbernatvasvari committed:
v1.6.3.3.10: I meant FORESTS. Last commit I meant FORESTS do not spawn any more on ocean. Mountains do. This patch increased forest spawn rate.

fabbernatvasvari committed:
v1.7: Big improvements on the desktop app. The desktop app's control flow now seamlessly integrates with the console app. Both apps can be controlled from console now. Also extracted a lot of repetitive code into methods. So overall the code quality is much better too.

fabbernatvasvari committed:
v1.7: - Feature update:  fixed some bugs
Big improvements on the desktop app. The desktop app's control flow now seamlessly integrates with the console app. Both apps can be controlled from console now. Also extracted a lot of repetitive code into methods. So overall the code quality is much better too.

Fabbernat committed:
Fix links to Polytopia frontend repository

Updated links to point to the correct frontend repository.

Fabbernat committed:
v1.8: added missing valid commands to the project

Fabbernat committed:
Merge remote-tracking branch 'origin/main'

Fabbernat committed:
Added images and descriptions for various map types including Drylands, Lakes, Continents, and Archipelagos.

Added images and descriptions for various map types including Drylands, Lakes, Continents, and Archipelagos. Updated contributions section with details on terrain generation and score counter modules.

Fabbernat committed:
Enhance README with new map types and contributions

Updated README.md with additional map types and contributions.

Fabbernat committed:
Revise map type descriptions in README.md

Updated descriptions for various map types and added insights.

Fabbernat committed:
Update headings for map types in README.md

Fabbernat committed:
v1.8.1: there is a known issue with water worlds

Fabbernat committed:
Merge remote-tracking branch 'origin/main'

Fabbernat committed:
v1.8.2: fix  yadakk tribe
